### PR TITLE
Fix/gcc8 compiler issues 1 and Host makefile extraneous character

### PR DESCRIPTION
**Pull request by frankdownunder**
I use G++ 8.3.0 which is fussy, it found some dodgy code in RCSwitch.
I also found an extraneous $ sign in the makefile that build the Host architecture

### CODE REVIEW

File: Sming/Arch/Host/sming.mk [hunk 1]
```diff
@@ -62,7 +62,7 @@ $(call UserLibPath,lwip): $(LWIP_BUILD_DIR)/Makefile
 
 $(LWIP_BUILD_DIR)/Makefile: $(LWIP_BASE)/lwip/.submodule | $(LWIP_BUILD_DIR)
 	$(Q) cd $(LWIP_BUILD_DIR); \
-	$(CMAKE) -DUSER_LIBDIR="$(SMING_HOME)/$(USER_LIBDIR)" $(CMAKE_OPTIONS) $(SMING_HOME$)/$(LWIP_BASE)/$(UNAME)
+	$(CMAKE) -DUSER_LIBDIR="$(SMING_HOME)/$(USER_LIBDIR)" $(CMAKE_OPTIONS) $(SMING_HOME)/$(LWIP_BASE)/$(UNAME)
```
Comment (slaff) [inline Sming/Arch/Host/sming.mk]: Nice catch!

File: Sming/Libraries/RCSwitch/RCSwitch.cpp [hunk 2]
```diff
@@ -285,7 +285,7 @@ char* RCSwitch::getCodeWordB(int nAddressCode, int nChannelCode, boolean bStatus
    
    const char* code[5] = { "FFFF", "0FFF", "F0FF", "FF0F", "FFF0" };
    if (nAddressCode < 1 || nAddressCode > 4 || nChannelCode < 1 || nChannelCode > 4) {
-    return '\0';
+    return nullptr;
```
Comment (slaff) [inline Sming/Libraries/RCSwitch/RCSwitch.cpp]: @mikee47 What do you think about this change?

Comment (mikee47) [reply Sming/Libraries/RCSwitch/RCSwitch.cpp]: Code will behave as it did before, which is probably wrong! It's used by `sendTriState()` which will then just segfault if it gets a NULL value. It should therefore be:

```
   if (nAddressCode < 1 || nAddressCode > 4 || nChannelCode < 1 || nChannelCode > 4) {
     sReturn[0] = '\0';
     return sReturn;
   }
```

File: Sming/Libraries/RCSwitch/RCSwitch.cpp [hunk 3]
```diff
@@ -348,7 +348,7 @@ char* RCSwitch::getCodeWordC(char sFamily, int nGroup, int nDevice, boolean bSta
   int nReturnPos = 0;
   
   if ( (byte)sFamily < 97 || (byte)sFamily > 112 || nGroup < 1 || nGroup > 4 || nDevice < 1 || nDevice > 4) {
-    return '\0';
+    return nullptr;
```
Comment (mikee47) [inline Sming/Libraries/RCSwitch/RCSwitch.cpp]: As for `getCodeWordB()`

File: Sming/Libraries/RCSwitch/RCSwitch.cpp [hunk 4]
```diff
@@ -418,7 +418,7 @@ char* RCSwitch::getCodeWordD(char sGroup, int nDevice, boolean bStatus){
         case 'D':
             sGroupCode = dec2binWcharfill(1, 4, 'F'); break;
         default:
-            return '\0';
+            return nullptr;
```
Comment (mikee47) [inline Sming/Libraries/RCSwitch/RCSwitch.cpp]: As for `getCodeWordB()`

File: Sming/Libraries/RCSwitch/RCSwitch.cpp [hunk 5]
```diff
@@ -437,7 +437,7 @@ char* RCSwitch::getCodeWordD(char sGroup, int nDevice, boolean bStatus){
         case 3:
             sDevice = dec2binWcharfill(1, 3, 'F'); break;
         default:
-            return '\0';
+            return nullptr;
```
Comment (mikee47) [inline Sming/Libraries/RCSwitch/RCSwitch.cpp]: As for `getCodeWordB()`

**Comment by mikee47** [comment]
`RCSwitch ` appears to have had some updates since this version, but curiously this bug hasn't been fixed. Perhaps it should be re-implemented as a patched submodule?

**Comment by slaff** [comment]
> RCSwitch appears to have had some updates since this version,

The latest update that I can find is from Nov 22, 2017. Is https://github.com/sui77/rc-switch the repo you are talking about or there is newer one?

**Comment by mikee47** [comment]
That's the one mentioned in the README.md, so I guess that's it. According to GIT the code hasn't been updated since it was added 1/5/2016.

**Comment by mikee47** [comment]
There are quite a few libraries which have been copied in rather than added as submodules, likely they're out of date too. I haven't done anything about this myself because it would require testing the changes, although we could just patch it and wait for the bug reports to come in...

**Comment by slaff** [comment]
> we could just patch it and wait for the bug reports to come in..

As long as the patch(es) fix issues in the RCSwitch code it would be best to send them upstream to the original project. Once they are merged I will take care to put RCSwitch as a submodule and provide 
a patch with our own changes.

**Comment by frankdownunder** [comment]
How long do you think it will take to get a proper fix? In the mean time, it doesnt compile for me!
Can we add it to the list of EXCLUDE_LIBRARIES or remove it or something? 
Or, since I havent changed the behaviour, implement my change as a temporary measure?

**Comment by slaff** [comment]
> How long do you think it will take to get a proper fix

That is the proper fix as suggested by Mike. 
```
if (nAddressCode < 1 || nAddressCode > 4 || nChannelCode < 1 || nChannelCode > 4) {
     sReturn[0] = '\0';
     return sReturn;
   }
```

@frankdownunder please, apply the fix it in this PR and I will take care to submit it as PR  to upstream.

**Pull request merged**